google/cloud/storage/internal/connection_impl_object_test.cc

```diff
@@ -370,6 +370,34 @@ TEST(StorageConnectionImpl, PatchObjectPermanentFailure) {
   EXPECT_THAT(permanent.captured_authority_options(), RetryLoopUsesOptions());
 }
 
+TEST(StorageConnectionImpl, RestoreObjectTooManyFailures) {
```

Comment:
nit: can we add a happy test case too?

Reply:
So looking at the other tests in this file it only tests the failure cases for connections. Just for the sake of continuity I'm not sure we want to add others here.